src/playbooks/with_build/vsp_health.yml

```diff
@@ -0,0 +1,188 @@
+  gather_facts: no
+  serial: 1
+  pre_tasks:
+    - name: Set upgrade flag
+      set_fact:
+        nuage_upgrade: true
+
+    - name: Check if prereq satisfied for upgrade
+      include_role:
+        name: common
+        tasks_from: check-upgrade-prereq
+      when: preupgrade is defined and preupgrade
+
+    - name: Set post-upgrade file name for VSTATs
+      set_fact:
+        report_filename: vstat_postupgrade_health_report
+
+    - name: Create preupgrade report
+      set_fact:
+        report_filename: vstat_preupgrade_health_report
+      when: preupgrade is defined and preupgrade
+  vars:
+    vstat_username: "{{ vstat_custom_username | default(vstat_default_username) }}"
+    vstat_password: "{{ vstat_custom_password | default(vstat_default_password) }}"
+  roles:
+    - vstat-health
+  remote_user: "{{ vstat_default_username }}"
+
+- hosts: vcins
+  gather_facts: no
+  serial: 1
+  pre_tasks:
+    - name: Set upgrade flag
+      set_fact:
+        nuage_upgrade: true
+
+    - name: Check if prereq satisfied for upgrade
+      include_role:
+        name: common
+        tasks_from: check-upgrade-prereq
+      when: preupgrade is defined and preupgrade
+
+    - name: Set post-upgrade file name for VCINs
+      set_fact:
+        report_filename: vcin_postupgrade_health_report
+
+    - name: Create preupgrade report
+      set_fact:
+        report_filename: vcin_preupgrade_health_report
+      when: preupgrade is defined and preupgrade
+  vars:
+    vsd_username: "{{ vsd_custom_username | default(vsd_default_username) }}"
+    vsd_password: "{{ vsd_custom_password | default(vsd_default_password) }}"
+  roles:
+    - vcin-health
+
+- hosts: vscs
+  gather_facts: no
+  serial: 1
+  connection: local
+  pre_tasks:
+    - name: Set upgrade flag
+      set_fact:
+        nuage_upgrade: true
+
+    - name: Check if prereq satisfied for upgrade
+      include_role:
+        name: common
+        tasks_from: check-upgrade-prereq
+      when: preupgrade is defined and preupgrade
+
+    - name: Set post-upgrade file name for VSCs on VSP
+      set_fact:
+        report_filename: vsp_postupgrade_health_report_for_vsc
+
+    - name: Create preupgrade report
+      set_fact:
+        report_filename: vsp_preupgrade_health_report_for_vsc 
+      when: preupgrade is defined and preupgrade
+  vars:
+    vsc_username: "{{ vsc_custom_username | default(vsc_default_username) }}"
+    vsc_password: "{{ vsc_custom_password | default(vsc_default_password) }}"
+  roles:
+    - vsc-health
+
+- hosts: primary_vsds
+  gather_facts: no
+  pre_tasks:
+    - name: Set upgrade flag
+      set_fact:
+        nuage_upgrade: true
+
+    - name: Check if prereq satisfied for upgrade
+      include_role:
+        name: common
+        tasks_from: check-upgrade-prereq
+      when: preupgrade is defined and preupgrade
+
+    - name: Set post-upgrade file name for primary_vsds
+      set_fact:
+        report_filename: vstat_postupgrade_health_report_for_vsd.txt

```

Comment:
I'm currently planning to pass in a variable for each report name from the pre/post upgrade health files. For example in preupgrade health, something like the following (except I'll substitute a proper name for each report):

- name: Run Preupgrade health
  import_playbook: "vsp_health.yml"
  vars:
    preupgrade: true
    vsd_report_name: preupgrade_health_report
    vsd_report_name: preupgrade_health_report
    vsd_report_name: preupgrade_health_report
    vsd_report_name: preupgrade_health_report
    vsd_report_name: preupgrade_health_report
    vsd_report_name: preupgrade_health_report
    vsd_report_name: preupgrade_health_report

and then instead of the two separate tasks to assign the pre/post report name, there will just be a single one to assign. For example:

    - name: Set report file name
      set_fact:
        report_filename: {{ vsd_report_name }}

is this what you had in mind, or would you suggest something else?

Reply:
Yes. That is correct.